BuildMasterAutomation/Functions/ConvertTo-BMOtterScriptExpression.ps1

```diff
@@ -0,0 +1,80 @@
+
+function ConvertTo-BMOtterScriptExpression
+{
+    <#
+    .SYNOPSIS
+    Converts a PowerShell object into an OtterScript expression.
+
+    .DESCRIPTION
+    The `ConvertTo-BMOtterScriptExpression` function takes a PowerShell object as an input and returns a representation
+    of the object in OtterScript. This function converts .NET IEnumerable and IDictionary  objects to OtterScript vector
+    and map types respectively.
+
+    .LINK
+    https://docs.inedo.com/docs/executionengine-otterscript-strings-and-literals
+
+    .EXAMPLE
+    ,@(1, 2, 3, 4) | ConvertTo-BMOtterScriptExpression
+
+    Demonstrates turning an IEnumerable of PowerShell integers into an array of OtterScript integers. Output will be
+    `@(1, 2, 3, 4)`
+
+    .EXAMPLE
+    @{ 'hello' = 'world'; 'goodbye' = 'world' } | ConvertTo-BMOtterScriptExpression
+
+    Demonstrates turning a PowerShell IDictionary into an OtterScript map. Output will be `%(hello: world, goodbye: world)`
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [Object] $Value
+    )
+
+    begin {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+    }
+
+    process {
+        $isDict = $Value -is [System.Collections.IDictionary]
+        $isList = $Value -is [System.Collections.IEnumerable] -and $Value -is [System.Collections.ICollection]
+
+        if (-not $isDict -and -not $isList)
+        {
+            return $Value

```

Comment:
This should write an error. The caller is expecting to get an OtterScript expression back as a string, so if we can't return that then we need to error.